### ISSUE TITLE
IKO - Ikoria: Lair of Behemoths Set Card Implementation Tracking

### ISSUE THREAD

**Issue by theelk801** [magefree/mage#6390]
Mark cards you're working on in the second post of this issue.
If you can't check things off, leave a comment instead.
Once the card is committed, check off the card here to show that it's finished.
Don't worry about adding new spoilers to this list, it's generated by a script.

[All Sets](https://github.com/magefree/mage/wiki/Set-implementation-list)

# Unimplemented Cards

- [ ] Not Done -- [Huntmaster Liger](https://scryfall.com/search?q=!"Huntmaster+Liger")
- [ ] Not Done -- [Illuna, Apex of Wishes](https://scryfall.com/search?q=!"Illuna,+Apex+of+Wishes")
- [ ] Not Done -- [Illuna, Apex of Wishes](https://scryfall.com/search?q=!"Illuna,+Apex+of+Wishes")
- [ ] Not Done -- [Illuna, Apex of Wishes](https://scryfall.com/search?q=!"Illuna,+Apex+of+Wishes")
- [ ] Not Done -- [Keruga, the Macrosage](https://scryfall.com/search?q=!"Keruga,+the+Macrosage")
- [ ] Not Done -- [Lukka, Coppercoat Outcast](https://scryfall.com/search?q=!"Lukka,+Coppercoat+Outcast")
- [ ] Not Done -- [Lukka, Coppercoat Outcast](https://scryfall.com/search?q=!"Lukka,+Coppercoat+Outcast")
- [ ] Not Done -- [Luminous Broodmoth](https://scryfall.com/search?q=!"Luminous+Broodmoth")
- [ ] Not Done -- [Vivien, Monsters' Advocate](https://scryfall.com/search?q=!"Vivien,+Monsters+Advocate")
- [ ] Not Done -- [Vivien, Monsters' Advocate](https://scryfall.com/search?q=!"Vivien,+Monsters+Advocate")
- [ ] Not Done -- [Zilortha, Strength Incarnate](https://scryfall.com/search?q=!"Zilortha,+Strength+Incarnate")
# Implemented Cards
 

- [x] Done -- [Gloom Pangolin](https://scryfall.com/search?q=!"Gloom+Pangolin")

**Comment by theelk801** [comment]
- [ ] In progress -- [Huntmaster Liger](https://scryfall.com/search?q=!"Huntmaster+Liger")
- [ ] In progress -- [Illuna, Apex of Wishes](https://scryfall.com/search?q=!"Illuna,+Apex+of+Wishes")
- [ ] In progress -- [Illuna, Apex of Wishes](https://scryfall.com/search?q=!"Illuna,+Apex+of+Wishes")
- [ ] In progress -- [Illuna, Apex of Wishes](https://scryfall.com/search?q=!"Illuna,+Apex+of+Wishes")
- [ ] In progress -- [Keruga, the Macrosage](https://scryfall.com/search?q=!"Keruga,+the+Macrosage")
- [ ] In progress -- [Lukka, Coppercoat Outcast](https://scryfall.com/search?q=!"Lukka,+Coppercoat+Outcast")
- [ ] In progress -- [Lukka, Coppercoat Outcast](https://scryfall.com/search?q=!"Lukka,+Coppercoat+Outcast")
- [ ] In progress -- [Luminous Broodmoth](https://scryfall.com/search?q=!"Luminous+Broodmoth")
- [ ] In progress -- [Vivien, Monsters' Advocate](https://scryfall.com/search?q=!"Vivien,+Monsters+Advocate")
- [ ] In progress -- [Vivien, Monsters' Advocate](https://scryfall.com/search?q=!"Vivien,+Monsters+Advocate")
- [ ] In progress -- [Zilortha, Strength Incarnate](https://scryfall.com/search?q=!"Zilortha,+Strength+Incarnate")

[Scryfall gallery of everything currently unimplemented](https://scryfall.com/search?q=!"Huntmaster%20Liger"or!"Illuna,%20Apex%20of%20Wishes"or!"Illuna,%20Apex%20of%20Wishes"or!"Illuna,%20Apex%20of%20Wishes"or!"Keruga,%20the%20Macrosage"or!"Lukka,%20Coppercoat%20Outcast"or!"Lukka,%20Coppercoat%20Outcast"or!"Luminous%20Broodmoth"or!"Vivien,%20Monsters'%20Advocate"or!"Vivien,%20Monsters'%20Advocate"or!"Zilortha,%20Strength%20Incarnate"+prefer%3Aoldest)

**Comment by theelk801** [comment]
Additional Task:

- [x] Implement Mutate #14900 
- [x] Implement Companion
- [x] Implement ability counters

**Comment by JayDi85** [comment]
@theelk801 don't add cards with mutate to completed list or add extra list for that cards in first post. It's confusing (cards "completed", but they don't).

**Comment by theelk801** [comment]
Yeah, I'm gonna add a filter like I had with Adventures, I just hadn't done it yet.

**Comment by emerald000** [comment]
Has someone started working on Mutate/Companion or should I do it?

**Comment by theelk801** [comment]
I haven't gotten to it yet but if you want to then go for it.

**Comment by emerald000** [comment]
Alright, I will start with the easy one (Companion) and see how I feel about Mutate afterwards.

**Comment by theelk801** [comment]
Yeah, that one's definitely easier but both seem like they're doable.

**Comment by Emigara** [comment]
I'm gona try code up the rest of the cards today.

**Comment by theelk801** [comment]
Awesome, just let me know which cards you're working on.

**Comment by Emigara** [comment]
tbh im struggling to get xmage working in eclipse so I can work on it im pretty new to github

**Comment by theelk801** [comment]
have you been following the guide on the wiki?

**Comment by Emigara** [comment]
ye I should get there

**Comment by Emigara** [comment]
Its not compiling in maven for some reason

**Comment by drowinternet** [comment]
Hi, just wanted to chime in. I tried to compile the client last week on Intellij and VSC and couldn't get it to finish- just like Emigara. Would love to know if there is a solution since I completed a couple cards but didn't submit because I couldn't play test them.

**Comment by theelk801** [comment]
what errors did it throw?

**Comment by JayDi85** [comment]
I'll try to implement mutate ability

**Comment by Emigara** [comment]
I'm trying to do it through the wizard for importing from git on eclipse

**Comment by Emigara** [comment]
if i can get this working i could have all the cards implemented tomorrow if you can get mutate implemented then get companion implemented.  Then we test and see if we can get ikoria onto xmage before it drops on arena

**Comment by Emigara** [comment]
I got in but cant playtest i can however create the implementations and send them to someone who can test

**Comment by theelk801** [comment]
Just implement the cards and open a pull request.

**Comment by drowinternet** [comment]
@theelk801  The client hangs when trying to load symbols when running in IntelligJ. It looked like the mavin install and running the server completed successfully.

**Comment by Emigara** [comment]
im working on mythos of snapdax atm and using tragic arrogance as a guide but wondering how on earth to change the effect to let you choose

**Comment by theelk801** [comment]
You can just have a different player make the choice

**Comment by theelk801** [comment]
@drowinternet have you followed the wiki guide? I think you need to allocate more ram

**Comment by drowinternet** [comment]
@theelk801 I'll have another look. I just updated the setting per the github guide

**Comment by Emigara** [comment]
[MythosOfSnapdax.txt](https://github.com/magefree/mage/files/4465489/MythosOfSnapdax.txt)

I havn't figured out yet how to commit to github so I uploaded code as txt file, could you test this for me until i figure out how to setup properly

**Comment by drowinternet** [comment]
just did a pull for 'back for more'. first card so will wait a bit to take another

**Comment by theelk801** [comment]
@Emigara you need to have the player choose the targets when the condition is true

**Comment by Emigara** [comment]
It is set up to do the normal thing when the condition isnt true

**Comment by theelk801** [comment]
it has the same player making the choice both times

**Comment by Emigara** [comment]
i must have messed it up, i'm currently in my bed as its 12am in scotland I will fix and resend 
tomorrow/

**Comment by theelk801** [comment]
No worries, try to see if you can get it committed so you can open a pull request

**Comment by drowinternet** [comment]
Pull request for 'monstrous step' submitted

**Comment by Emigara** [comment]
how close are we do you reckon until we can release ikoria to live?

**Comment by Emigara** [comment]
[MythosOfSnapdax.txt](https://github.com/magefree/mage/files/4466146/MythosOfSnapdax.txt)
I think i fixed it

**Comment by Emigara** [comment]
I submitted a commit

**Comment by Emigara** [comment]
@JayDi85 how close are you to getting mutate implemented as I can implement the rest of the mutate cards when the mechanic is in place

**Comment by JayDi85** [comment]
@Emigara I'm still working on mutate only ability, not cards. Sync with @theelk801 about that cards implementation (with disabled mutate).

**Comment by Emigara** [comment]
ok

**Comment by theelk801** [comment]
There are only three mutate cards left that need to be done. Two of them are just a little complicated so I was putting them off but the third one needs mutate to actually be implemented in order to be done.

**Comment by Emigara** [comment]
Did my commit from my eclipse client go through for Mythos of Snapdax as I had i believe fixed the mistake i made.

**Comment by theelk801** [comment]
Are you trying to push directly to the repo? You need to make a fork and push to that, then open a pull request.

**Comment by Emigara** [comment]
that explains why im struggling I will set that up later.  I am so noob when it comes to github lol.

**Comment by theelk801** [comment]
It's alright, we've all been there.

**Comment by drowinternet** [comment]
I still haven't implemented 'back for more' and might not be able to get to it since the week is starting. please feel free to take over the implementation if you get the time.

**Comment by CCTV-1** [comment]
[Blitz of the Thunder-Raptor](https://scryfall.com/card/iko/109/blitz-of-the-thunder-raptor)  target will is creature or planeswalker,but https://github.com/magefree/mage/blob/9979f76542374686a1e9fa160826ea8e356919ba/Mage.Sets/src/mage/cards/b/BlitzOfTheThunderRaptor.java#L33 current target is player or planeswalker

**Comment by CCTV-1** [comment]
[Wingspan Mentor](https://scryfall.com/card/iko/72/wingspan-mentor) activated ability should be 2U,
https://github.com/magefree/mage/blob/9979f76542374686a1e9fa160826ea8e356919ba/Mage.Sets/src/mage/cards/w/WingspanMentor.java#L57 current is 2R.

**Comment by CCTV-1** [comment]
[Call of the Death-Dweller](https://scryfall.com/card/iko/78/call-of-the-death-dweller) AI can over the cmc limit. 

game log:[20200413_192837_Two Player Duel_1.txt](https://github.com/magefree/mage/files/4469337/20200413_192837_Two.Player.Duel_1.txt)

**Comment by Emigara** [comment]
uploading mythos of snapdax shortly managed to create a fork

**Comment by Emigara** [comment]
sent pull request for Mythos of Snapdax

**Comment by Emigara** [comment]
Curious as to how far along mutate is?

**Comment by Emigara** [comment]
gona work on valiant rescuer today

**Comment by Emigara** [comment]
rescuer pull request up

**Comment by Emigara** [comment]
Rielle, The Everwise is next

**Comment by Emigara** [comment]
any cards exist that has draw equal to amount of cards discarded so i can finish up implementing rielle

**Comment by Emigara** [comment]
managed to finish up valiant rescuer but wanna have rielle get put in on the same pull request

**Comment by theelk801** [comment]
There isn't really another card that does that at the moment

**Comment by Emigara** [comment]
got rielle and valiant rescuer pulled btw

**Comment by htrajan** [comment]
working on Zilortha

**Comment by htrajan** [comment]
Zilortha completed

**Comment by beyondthegrave88** [comment]
where or how do i program a card into xmage  i would love to learn as we are missing false dawn from apocalypse and i haven't noticed any other missing cards but would like to go through and see if there is something i can do to help i really have never programed anything but am a fast learner and would be very happy to help thanks.

**Comment by htrajan** [comment]
> 
> 
> where or how do i program a card into xmage i would love to learn as we are missing false dawn from apocalypse and i haven't noticed any other missing cards but would like to go through and see if there is something i can do to help i really have never programed anything but am a fast learner and would be very happy to help thanks.

check out http://github.com/magefree/mage/wiki/Developer-Getting-Started

**Comment by Emigara** [comment]
ye its taking me a while to learn how xmage works plus arena just got ikoria so i was held up with that too.  gona probably try fix the 2 cards im working tomorrow but if yous have managed to fix what i have there just now feel free to implement it for me.

**Comment by superschure** [comment]
Thank you for all your hard and quick work, especially during these difficult times. I wanted to inform you that while using "Zirda, the Dawnwaker" I noticed that instead of ability costs being reduced by 2, casting costs are reduced by 2.

**Comment by beyondthegrave88** [comment]
htrajan thanks man im working on the downloads now im going to see if i can be of some help on the next set as this will be my first endevor on coding hopefully i don't code grey goo or some crazy shit, lol

**Comment by SteToe** [comment]
Hey everyone! Thanks for your hard and wonderful work! I don't understand the first thing about coding, so there isn't much support I can offer. But I found this:

Put a menace counter ("Blood Curdle") onto my "Boot Nipper" (with Lifelink counter). On the following turn my opponent was able to block the little guy with their sole "Exhuberant Wolfbear". The lifelink did work, the menace obviously not.

Keep it up! And stay safe and sane everyone!

**Comment by theelk801** [comment]
Thanks, both bugs have been fixed.

**Comment by SteToe** [comment]
"Mythos of Snapdax": _Each player chooses an artifact, a creature, an enchantment, and a planeswalker from among the nonland permanents they control, then sacrifices the rest. If {B}{R} was spent to cast this spell, you choose the permanents for each player instead._
The mouseover text reads "you choose" in both parts though and it plays like that as well: the player casting the spell gets to choose wether the requirement is met or not.

**Comment by edidavide** [comment]
The Ozolith is bugged in the beta server, when you select a creature you want to transfer counters to, counters are just added to the creature but not removed from the ozolith itself and can be used in following turns

**Comment by HenriLamarre** [comment]
Hey guys! Thanks for the awesome work you guys are doing! I wish I knew java and could help you coding but for now imma stick to playtesting. Heres a short list of bugs I found:
-Couldnt cast survivor's bond with a human in the yard
-Lukka's first ability is wrong
-For some reason, Gyruda costed me 4 to cast in a game and 6 to cast in another.
(Also, the opponents yard does not pop up when the ability resolves and we dont have the option of selecting cards from that yard (but that might have been an oversight from me since maybe no cards with even mana cost were put there that way))
-Fire prophecy lets you keep the card in your hand

**Comment by SacnothXIII** [comment]
Hello, first of all, thanks for your amazing work.
I've been trying to test Ikoria but server keeps telling me my cards are illegal and when I change to other version of the same card it points another one. Now I am stuck with a card with only one version. I guess it's a problem with the set or install. 
I have traditional and Beta XMage installed. Different disks, different folders. 
Normal works just fine. Can somebody help me?

**Comment by gp66** [comment]
First off, amazing project! 

I don't know Java so all I can do is report minor bugs.

Stormwild Capridor:
I targeted my own attacking Stormwild Capridor with Shock. 
The non-combat damage was not prevented and the two +1/+1 counters were not added.
The card's image reads 'noncombat damage', xmage's popup text reads 'combat damage'
On a subsequent attack, Stormwild Capridor was blocked and the damage was prevented 
and the +1/+1 counters were added for combat damage.

Skyknight Vanguard:
Skyknight Vanguard was attacking due to 'Winota, Joiner of Forces' triggered ability 
and the 1/1 white Soldier creature token was not added to the battlefield tapped and attacking.

Thank you for all the hard work!

![Screenshot (242)](https://user-images.githubusercontent.com/22893068/80164049-8ff85e00-859d-11ea-9435-f688acbfdf2b.png)

**Comment by theelk801** [comment]
Stormwild Capridor has already been fixed, and the Winota interaction is correct as Skyknight Vanguard never actually attacked

**Comment by beyondthegrave88** [comment]
when will this be on all xmage?

**Comment by halpenstance** [comment]
It seems like Valiant Rescuer https://scryfall.com/card/iko/36/valiant-rescuer appears to be missing.

**Comment by Jstiller97** [comment]
It seems like the beta server is down, is there a problem with it? Was trying to bug test the new cards but to no avail.

**Comment by Jstiller97** [comment]
I can't seem to download ikoria images, any suggestions?

**Comment by JayDi85** [comment]
@Jstiller97 ensure that you use [beta client](http://xmage.today/), open download images dialog, choose scryfall source (it's selected by default) and choose IKO set from list. That's all.

![shot_200429_214231](https://user-images.githubusercontent.com/8344157/80628417-5d010f00-8a62-11ea-936a-b6bb0194f28c.png)

**Comment by Rastael** [comment]
Heyho :)

I don't want to stress you guys or be unpatient. I'm just curious, if there are any news of implemnting the mutate-feature to Xmage. I'd really like to draft that format during quaranteen and if you need more time, I will try out Arena (even if I don't like it). ^^

**Comment by JayDi85** [comment]
@Rastael soon

**Comment by superwill123** [comment]
Not to sound rude (I HIGHLY appreciate all that you do for this FTP program), but if programming in the mutate mechanic is going to take so long, would it be better to post an update with all the new Ikoria cards sans the mutate cards, like you did with C19 with cards like K'rrik?

**Comment by footnmouth5** [comment]
@superwill123 They have implemented available IKO cards into the beta client. Play on the beta if you want to play with some IKO cards.

**Comment by DominusMaximus** [comment]
Shark Typhoon's trigger always creates 0/0 Sharks regardless of the noncreature spell's converted mana cost.

**Comment by fearphage** [comment]
> They have implemented available IKO cards into the beta client. Play on the beta if you want to play with some IKO cards.

Pardon my ignorance, but how does one do this? When I try to import IKO cards into decks via the deck editor it says they cannot be found. So how do I play with them if I can't put them into decks?

**Comment by Rastael** [comment]
Did you use the beta server?

**Comment by fearphage** [comment]
Tried numerous times. I've got the server version mismatch error numerous times as well as just not being able to connect. I'll just wait patiently I suppose.

I did find the beta version of the client which let me import the new IKO cards though... so part of the way there.

**Comment by DominusMaximus** [comment]
I put an Agent of Treachery on the stack. In response, my opponent used Lurrus of the Dream-Den's ability to cast a Spectral Sailor with flash from their graveyard. After gaining control of the Lurrus, I was unable to cast a Gilded Goose from my own graveyard. It seems it was counting the Spectral Sailor my opponent had cast as the one permanent spell per turn. On subsequent turns, Lurrus worked correctly under my control.

**Comment by superwill123** [comment]
> @superwill123 They have implemented available IKO cards into the beta client. Play on the beta if you want to play with some IKO cards.

While this IS an option, and I mean no disrespect in what I am about to say, but in all honesty, if one doesn't want to play on a beta client but wants to play with new cars that are ready for release, they shouldn't have be forced to download something they don't want to download or have to wait past the start of spoiler season of the next set to play cards that have existed in paper. So unless I'm mistaken in some way or another (if so, please do correct me), it should be possible to add all the non-mutate cards into the non-beta client and release it like that, right?

TL;DR claiming that you should jump to another version of the game just because you want to wait for all the cards to be finished is not a good business practice

**Comment by Toshenko** [comment]
hey i love xmage, i'm supporting and donate for this wonderful project. Please communicate on what's happening in studio, i can't wait playing my new modern and impossible to charge a list on the beta...

**Comment by DominusMaximus** [comment]
> TL;DR claiming that you should jump to another version of the game just because you want to wait for all the cards to be finished is not a good business practice

The Beta serves as a first run of newly created and implemented cards and mechanics. It allows players to report bugs, and programmers to correct errors or interactions with quirks! (before landing on the "stable" release.) Once downloaded, the Beta easily updates itself, and with portable Java and Java Launcher (portableapps.com), one can create a portable Beta with all images that easily updates itself. And if you add a 2nd folder containing the stable release, you can simply choose which portable client to use.

**Comment by htrajan** [comment]
Any way I can help out on mutate?

**Comment by superwill123** [comment]
> The Beta serves as a first run of newly created and implemented cards and mechanics. It allows players to report bugs, and programmers to correct errors or interactions with quirks! (before landing on the "stable" release.) Once downloaded, the Beta easily updates itself, and with portable Java and Java Launcher (portableapps.com), one can create a portable Beta with all images that easily updates itself. And if you add a 2nd folder containing the stable release, you can simply choose which portable client to use.

Oh, I'm not doubting the capabilities of the beta server. I'm just saying that you shouldn't force people to download an entirely new client just to play with cards that have been out for MONTHS when they are clearly ready to be released to the public in the non-beta version of the game.

**Comment by sirjonas** [comment]
Any news about the cards going live? I saw that there are only 2 cards left.

**Comment by JayDi85** [comment]
@sirjonas new release will be in 1 day. C20 plus IKO (without mutate cards).

**Comment by sirjonas** [comment]
> @sirjonas new release will be in 1 day. C20 plus IKO (without mutate cards).

OH REALLY? Thats great news! Where did you get that info? Thanks for sharing

**Comment by JayDi85** [comment]
@sirjonas I'm preparing what's new docs and some little commits now. After that new release will be ready.

**Comment by fearphage** [comment]
> I'm preparing what's new docs and some little commits now. After that new release will be ready.

I just checked and I still can't important IKO cards on the stable server. Is there something I can watch/follow/subscribe to to see when this lands?

**Comment by superwill123** [comment]
> new release will be in 1 day.

![image](https://user-images.githubusercontent.com/64988489/84454080-9911bd00-ac27-11ea-8141-a61ab811a40d.png)

**Comment by Kovo** [comment]
Still no release?

**Comment by superwill123** [comment]
> Still no release?

Still no release. My guess is that they're implementing the new "commander dies" rules, which I can respect

**Comment by sirjonas** [comment]
Any news? It would be Nice to know whats happening.

**Comment by sirjonas** [comment]
Any news?

**Comment by sumsarz** [comment]
It's not ready, and will be released when it's released.

There, happy?

**Comment by edidavide** [comment]
> It's not ready, and will be released when it's released.
> 
> There, happy?

People asking for info on a new release is actually a good thing for the project, it means people are interested in this.
Would xmage even exist without people being interested in using it ?

Besides, it's been released 30 mins ago :D !

Many thanks to the devs !

**Comment by JayDi85** [comment]
I apologize for the delay in the release. Now everything is ready and the new version is installed on the server.

**Comment by ghvanderweg** [comment]
@JayDi85 any word on when mutate will be ready for release? :)

**Comment by LotusWalker** [comment]
Any thoughts on when Mutate might be ready for release?

**Comment by JayDi85** [comment]
Nope, it's under development. You can find more info and testing version in [forum's topic](https://www.slightlymagic.net/forum/viewtopic.php?f=116&p=244918).

**Comment by jbureau88** [comment]
Seems like mutate progress was pretty much lost? The post on [slightlymagic](https://www.slightlymagic.net/forum/viewtopic.php?f=116&t=29937&sid=2b814d5d54a2e19b838ecc7a298b17b0&start=30) had a link to the latest code base but it is no longer a valid link and seems like it has been 2 years without any activity or progress on the mutate ability sadly.

Any way we can start working on it again  ?

**Comment by Grath** [comment]
There's #8733 which has substantial open issues still, and has major unresolved implications around targeting, attaching cards, handling effects... Feel free to pick up working on that pull request to see if you can make Mutate actually work correctly.

**Comment by xenohedron** [comment]
Mutate is implemented by #14900, so this can finally be closed
